source/WebConfig/cosa_webconfig_dml.c

```diff
@@ -256,9 +267,100 @@ BOOL isValidUrl
 	WebcfgDebug("Validate URL %s\n", pUrl);
 	if(strstr(pUrl, "https") == NULL)
 	{
+		if(1 == get_supplementary_flag() && strncmp(pUrl,"NULL", strlen("NULL")) == 0)
+		{
+			return TRUE;
+		}
 		WebcfgError("Invalid URL, HTTPS is only allowed\n");
 		return FALSE;
 	}
 	return TRUE;
 }
 
+/***********************************************************************
+
+ APIs for Object:
+
+    X_RDK_WebConfig.SupplementaryServiceUrls.
+
+    *  Supplementary_Service_Urls_GetParamStringValue
+    *  Supplementary_Service_Urls_SetParamStringValue
+
+***********************************************************************/
+
+BOOL
+Supplementary_Service_Urls_SetParamStringValue
+    (
+        ANSC_HANDLE                 hInsContext,
+        char*                       ParamName,
+        char*                       strValue
+    )
+{
+	WebcfgDebug(" %s : ENTER \n", __FUNCTION__ );
+	RFC_ENABLE=Get_RfcEnable();
+	if(!RFC_ENABLE)
+	{
+		WebcfgError("%s RfcEnable is disabled so, %s SET failed\n",__FUNCTION__,ParamName);
+		return FALSE;
+	}
+
+	if( AnscEqualString(ParamName, "Telemetry", TRUE))
+        {
+		WebcfgDebug("strValue is%s-url\n", strValue);
+		set_supplementary_flag(1);
+		if(isValidUrl(strValue) == TRUE || strncmp(strValue,"NULL",strlen("NULL")) == 0)
```

Comment:
Is this strncmp "OR" condition again required here , as it is already handled in isValidUrl function ?